### PR TITLE
Cypress test. Group features.

### PR DESCRIPTION
**Pull request by dvkruchinin**
Cypress test. Group features.

### Checklist

- [x] I submit my changes into the `develop` branch
- [ ] I have added description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2020 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: tests/cypress/integration/actions_tasks_objects/case_15_group_features.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+            cy.createRectangle(createRectangleTrack2Points);
+            cy.createRectangle(createRectangleTrack2PointsSecond);
+        });
+        it('Set option "Color by" to "Group".', () => {
+            cy.changeAppearance('Group');
+            cy.get('.cvat_canvas_shape').then($listCanvasShapes => {
+                for (let i=0; i<$listCanvasShapes.length; i++) {
+                    cy.get($listCanvasShapes[i]).should('have.css', 'fill').then($fill => {
+                        // expected rgb(224, 224, 224) to equal rgb(224, 224, 224)
+                        expect($fill).to.be.equal($fill);
```
Comment (bsekachev) [inline tests/cypress/integration/actions_tasks_objects/case_15_group_features.js:69]: ``expect($fill).to.be.equal($fill);``? How does it work?

File: tests/cypress/integration/actions_tasks_objects/case_15_group_features.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+            for (let groupedShape of ['#cvat_canvas_shape_1', '#cvat_canvas_shape_2']) {
+                cy.get(groupedShape).should('have.css', 'fill').then($fillGroupedShape => {
+                    // expected rgb(250, 50, 83) to not equal rgb(224, 224, 224)
+                    expect($fillGroupedShape).to.not.equal(fillChangeAppearance);
+                    // expected rgb(250, 50, 83) to equal rgb(250, 50, 83)
+                    expect($fillGroupedShape).to.be.equal($fillGroupedShape);
+                    fillGroupedShape = $fillGroupedShape;
+                });
+            }
+            for (let objectSideBarShape of ['#cvat-objects-sidebar-state-item-1', '#cvat-objects-sidebar-state-item-2']) {
```
Comment (bsekachev) [inline tests/cypress/integration/actions_tasks_objects/case_15_group_features.js:98]: for (const of ..

File: tests/cypress/integration/actions_tasks_objects/case_15_group_features.js [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        points: 'By 2 Points',
+        type: 'Track',
+        switchLabel: false,
+        firstX: createRectangleTrack2Points.firstX + 300,
+        firstY: createRectangleTrack2Points.firstY,
+        secondX: createRectangleTrack2Points.secondX + 300,
+        secondY: createRectangleTrack2Points.secondY,
+    };
+
+    let fillChangeAppearance = '';
```
Comment (bsekachev) [inline tests/cypress/integration/actions_tasks_objects/case_15_group_features.js:48]: Let's rename to ``defaultGroupColor``

File: tests/cypress/integration/actions_tasks_objects/case_15_group_features.js [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    let fillGroupedShape = '';
```
Comment (bsekachev) [inline tests/cypress/integration/actions_tasks_objects/case_15_group_features.js:49]: shapesGroupColor

File: tests/cypress/integration/actions_tasks_objects/case_15_group_features.js [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+    let fillGroupedTrack = '';
```
Comment (bsekachev) [inline tests/cypress/integration/actions_tasks_objects/case_15_group_features.js:50]: tracksGroupColor

File: tests/cypress/integration/actions_tasks_objects/case_15_group_features.js [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+            for (let groupedTrack of ['#cvat_canvas_shape_3', '#cvat_canvas_shape_4']) {
+                cy.get(groupedTrack).should('have.css', 'fill').then($fillGroupedTrack => {
+                    // expected rgb(250, 50, 83) to not equal rgb(224, 224, 224)
+                    expect($fillGroupedTrack).to.not.equal(fillChangeAppearance);
+                    // expected rgb(250, 50, 83) to equal rgb(250, 50, 83)
+                    expect($fillGroupedTrack).to.be.equal($fillGroupedTrack);
+                    fillGroupedTrack = $fillGroupedTrack;
+                });
+            }
+            for (let objectSideBarTrack of ['#cvat-objects-sidebar-state-item-3', '#cvat-objects-sidebar-state-item-4']) {
```
Comment (bsekachev) [inline tests/cypress/integration/actions_tasks_objects/case_15_group_features.js:122]: for (const of

File: tests/cypress/integration/actions_tasks_objects/case_15_group_features.js [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+            for (let shapeToGroup of ['#cvat_canvas_shape_1', '#cvat_canvas_shape_2']) {
+                cy.get(shapeToGroup).click();
+            }
+            cy.get('.cvat-group-control').click();
+            for (let groupedShape of ['#cvat_canvas_shape_1', '#cvat_canvas_shape_2']) {
+                cy.get(groupedShape).should('have.css', 'fill').then($fillGroupedShape => {
+                    // expected rgb(250, 50, 83) to not equal rgb(224, 224, 224)
+                    expect($fillGroupedShape).to.not.equal(fillChangeAppearance);
+                    // expected rgb(250, 50, 83) to equal rgb(250, 50, 83)
+                    expect($fillGroupedShape).to.be.equal($fillGroupedShape);
```
Comment (bsekachev) [inline tests/cypress/integration/actions_tasks_objects/case_15_group_features.js:94]: ``expect($fillGroupedShape).to.be.equal($fillGroupedShape);``?

File: tests/cypress/integration/actions_tasks_objects/case_15_group_features.js [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+            for (let trackToGroup of ['#cvat_canvas_shape_3', '#cvat_canvas_shape_4']) {
+                cy.get(trackToGroup).click();
+            }
+            cy.get('.cvat-group-control').click();
+            for (let groupedTrack of ['#cvat_canvas_shape_3', '#cvat_canvas_shape_4']) {
+                cy.get(groupedTrack).should('have.css', 'fill').then($fillGroupedTrack => {
+                    // expected rgb(250, 50, 83) to not equal rgb(224, 224, 224)
+                    expect($fillGroupedTrack).to.not.equal(fillChangeAppearance);
+                    // expected rgb(250, 50, 83) to equal rgb(250, 50, 83)
+                    expect($fillGroupedTrack).to.be.equal($fillGroupedTrack);
```
Comment (bsekachev) [inline tests/cypress/integration/actions_tasks_objects/case_15_group_features.js:118]: ``expect($fillGroupedTrack).to.be.equal($fillGroupedTrack);``?

**Review by bsekachev — CHANGES_REQUESTED** [review]

File: tests/cypress/integration/actions_tasks_objects/case_15_group_features.js (hunk 1 above)
Comment (dvkruchinin) [reply tests/cypress/integration/actions_tasks_objects/case_15_group_features.js]: Yes, this is not how it should be. Thank you, corrected. Here I make a check that the checked parameter is the same for all elements in the loop

File: tests/cypress/integration/actions_tasks_objects/case_15_group_features.js (hunk 2 above)
Comment (dvkruchinin) [reply tests/cypress/integration/actions_tasks_objects/case_15_group_features.js]: Replaced.

File: tests/cypress/integration/actions_tasks_objects/case_15_group_features.js (hunk 3 above)
Comment (dvkruchinin) [reply tests/cypress/integration/actions_tasks_objects/case_15_group_features.js]: Done.

File: tests/cypress/integration/actions_tasks_objects/case_15_group_features.js (hunk 4 above)
Comment (dvkruchinin) [reply tests/cypress/integration/actions_tasks_objects/case_15_group_features.js]: Done.

File: tests/cypress/integration/actions_tasks_objects/case_15_group_features.js (hunk 5 above)
Comment (dvkruchinin) [reply tests/cypress/integration/actions_tasks_objects/case_15_group_features.js]: Done.

File: tests/cypress/integration/actions_tasks_objects/case_15_group_features.js (hunk 6 above)
Comment (dvkruchinin) [reply tests/cypress/integration/actions_tasks_objects/case_15_group_features.js]: Replaced.

File: tests/cypress/integration/actions_tasks_objects/case_15_group_features.js (hunk 7 above)
Comment (dvkruchinin) [reply tests/cypress/integration/actions_tasks_objects/case_15_group_features.js]: As well as for ```$fill```

File: tests/cypress/integration/actions_tasks_objects/case_15_group_features.js (hunk 8 above)
Comment (dvkruchinin) [reply tests/cypress/integration/actions_tasks_objects/case_15_group_features.js]: As well as for ```$fill```

**Review by bsekachev — APPROVED** [review]

**Pull request merged**